services/web-server/src/graphql/Hooks.graphql

```diff
@@ -72,11 +72,17 @@ type DeleteHook {
   hookId: ID
 }
 
+input HookBinding {
+  exchange: String!
+  pattern: String!
```

Comment:
I think this property needs to be renamed to be in line with what the hook client expects.

Reply:
The actual mutations are happening in `services/web-server/src/resolvers/hooks.js`.

Reply:
Defintion of a hook: https://github.com/taskcluster/taskcluster/blob/87facc8e7fc4580440bf730e2a0f5e080aba3154/services/hooks/schemas/v1/hook-definition.yml

Definition of a binding: https://github.com/taskcluster/taskcluster/blob/87facc8e7fc4580440bf730e2a0f5e080aba3154/services/hooks/schemas/v1/bindings.yml

Reply:
> I think this property needs to be renamed to be in line with what the hook client expects.

I have updated this

Reply:
Let me checkin my latest changes

Reply:
After saving I am getting success message, how to verify it that it has saved correctly?

Reply:
After saving, reloading the page should display the existent bindings that were saved earlier.

Reply:
On page load this query is getting fired

```
"query Hook($hookGroupId: ID!, $hookId: ID!) {
  hook(hookGroupId: $hookGroupId, hookId: $hookId) {
    hookGroupId
    hookId
    schedule
    metadata {
      name
      description
      owner
      emailOnError
      __typename
    }
    status {
      nextScheduledDate
      lastFire {
        ... on HookSuccessfulFire {
          time
          taskId
          __typename
        }
        ... on HookFailedFire {
          time
          error
          __typename
        }
        ... on NoFire {
          result
          __typename
        }
        __typename
      }
      __typename
    }
    triggerSchema
    task
    __typename
  }
  hookLastFires(hookGroupId: $hookGroupId, hookId: $hookId) {
    taskId
    firedBy
    taskCreateTime
    result
    error
    __typename
  }
}
"
```
And I don't see `bindings` field getting queried in this.
I tried adding bindings in this query but its giving 404 after adding bindings in the above query

Reply:
Ah, you would need to add it in web-server:

```diff
diff --cc ui/src/views/PulseMessages/index.jsx
index 85010860d,4371e89d6..000000000
--- a/ui/src/views/PulseMessages/index.jsx
+++ b/ui/src/views/PulseMessages/index.jsx
diff --git a/services/web-server/src/graphql/Hooks.graphql b/services/web-server/src/graphql/Hooks.graphql
index 1c59765f3..040ca0f52 100644
--- a/services/web-server/src/graphql/Hooks.graphql
+++ b/services/web-server/src/graphql/Hooks.graphql
@@ -51,6 +51,7 @@ type HookStatus {
 type Hook {
   hookGroupId: ID!
   hookId: ID!
+  bindings: [HookBinding]
   metadata: HookMetadata!
   schedule: [String]!
   task: JSON!
@@ -72,7 +73,12 @@ type DeleteHook {
   hookId: ID
 }
 
-input HookBinding {
+type HookBinding {
+  exchange: String!
+  routingKeyPattern: String!
+}
+
+input HookBindingInput {
   exchange: String!
   routingKeyPattern: String!
 }
@@ -82,7 +88,7 @@ input HookInput {
   schedule: [String]
   expires: String
   deadline: String
-  bindings: [HookBinding]
+  bindings: [HookBindingInput]
   task: JSON!
   triggerSchema: JSON
 }
```

Reply:
I am getting insufficient scopes error again

```
{"errors":[{"message":"Client ID mozilla-auth0/oauth2|firefoxaccounts|967134a2ed244f61b75935fc10b9119c|arshadkazmi42@gmail.com/taskcluster-web-server does not have sufficient scopes and is missing the following scopes:\n\n```\n{\n  \"AllOf\": [\n    \"hooks:modify-hook:garbage/arshad\",\n    \"assume:hook-id:garbage/arshad\"\n  ]\n}\n```\n\n
```

can you add access for this client id to add bindings? Or can you let me know how can I do this, if my client id changes

Reply:
Done.